src/ralph/admin/autocomplete.py

```diff
@@ -140,14 +86,70 @@ def get_queryset(self, user):
 
         my_urls = [
             url(
-                r'^autocomplete/suggest/$',
-                List.as_view(),
-                name='{}_{}_autocomplete_suggest'.format(*params)
-            ),
-            url(
                 r'^autocomplete/details/$',
                 Detail.as_view(),
                 name='{}_{}_autocomplete_details'.format(*params)
             ),
         ]
         return my_urls + urls
+
+
+class AutocompleteList(SuggestView):
+    limit = 10
+    model = None
+
+    def dispatch(self, request, *args, **kwargs):
+        model = get_model(kwargs['app'], kwargs['model'])
+        self.field = model._meta.get_field(kwargs['field'])
+        self.model = self.field.rel.to
+        self.query = request.GET.get(QUERY_PARAM, None)
+        if not self.query:
+            return HttpResponseBadRequest()
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_base_ids(self, model, value):
+        """
+        Return IDs for related models.
+        """
+        search_fields = ralph_site._registry[model].search_fields
+        query_filters = [
+            Q(**{'{}__icontains'.format(field): value})
+            for field in search_fields
+        ]
+        if issubclass(model, PermissionsForObjectMixin):
+            queryset = model._get_objects_for_user(
+                self.request.user, model.objects
+            )
+        else:
+            queryset = model.objects
+        return queryset.filter(
+            *query_filters
+        )[:self.limit].values_list('pk', flat=True)
+
+    def get_queryset(self, user):
+        search_fields = ralph_site._registry[self.model].search_fields
+        queryset = self.model._default_manager.all()
```

Comment:
could you add ability to easily overwrite queryset? I'll need this in https://github.com/allegro/ralph/pull/2033/files#diff-c920c22878885266492749084f889519R63